### ISSUE TITLE
iris.load_cube attempts to merge twice

### ISSUE THREAD

**Issue by rcomer** [SciTools/iris#3334]
Suppose we have a set of cubes with matching dim coords but mismatched aux coords:

```python
import iris

orig_cube = iris.cube.Cube(range(4), long_name='foo')
orig_cube.add_dim_coord(iris.coords.DimCoord(range(4), long_name='bar'), 0)

cubelist = iris.cube.CubeList()

for i in range(5):
    new_cube = orig_cube.copy()
    new_cube.add_aux_coord(
        iris.coords.DimCoord(i + 1, standard_name='realization'))
    new_cube.add_aux_coord(
        iris.coords.DimCoord(range(i // 3, i // 3 + 4), long_name='baz'), 0)
    cubelist.append(new_cube)
```
These will merge into 2 cubes:

```python
print(cubelist.merge())

0: foo / (unknown)                     (realization: 3; bar: 4)
1: foo / (unknown)                     (realization: 2; bar: 4)
```

If I try to merge to a single cube, I get a useful error about the mismatched coord:

```python
print(cubelist.merge_cube())

iris.exceptions.MergeError: failed to merge into a single cube.
  Coordinates in cube.aux_coords (non-scalar) differ: baz.
```

However, if each of these cubes is in its own file and I try to load them as one cube, I get a not so helpful error:

```python
for i, cube in enumerate(cubelist):
    iris.save(cube, 'test_{}.nc'.format(i))

iris.load_cube('test_?.nc')

iris.exceptions.ConstraintMismatchError: failed to merge into a single cube.
  cube.shape differs: (3, 4) != (2, 4)
```

It appears that `iris.load_cube` tries to re-merge a cubelist that has already been merged.  Could the first merge be skipped?

**Comment by rcomer** [comment]
Found some history: #1026

Originally there was one merge, but the second was added to give a more informative error.

**Comment by bjlittle** [comment]
To be honest, the more informative error message is:
```python
ConstraintMismatchError: failed to merge into a single cube.
  Coordinates in cube.aux_coords (non-scalar) differ: baz.
```

**Comment by rcomer** [comment]
@bjlittle yes sorry, I wan't clear there.  Originally it just did `merge` and then, if the merged cubelist had more than one cube, raised an error to say "can't do it".  The `merge_cube` was added in so that _in most circumstances_ you would get something more informative, e.g. "can't do it because different stash codes".

**Comment by rcomer** [comment]
Addressed by #3335.
